spec/services/upcoming_changes/promoter_spec.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+RSpec.describe UpcomingChanges::Promoter do
+  describe ".call" do
+    subject(:result) { described_class.call(params:, **dependencies) }
+
+    fab!(:admin)
+    let(:current_change_status) { :beta }
+    let(:dependencies) { { guardian: } }
+    let(:guardian) { admin.guardian }
+    let(:params) { { setting_name: :enable_upload_debug_mode, promotion_status_threshold: } }
+
+    before do
+      mock_upcoming_change_metadata(
+        {
+          enable_upload_debug_mode: {
+            impact: "other,developers",
+            status: current_change_status,
+            impact_type: "other",
+            impact_role: "developers",
+          },
+        },
+      )
+    end
+
+    context "when the upcoming change does not meet the status promotion criteria" do
+      let(:promotion_status_threshold) { :stable }
+      it { is_expected.to fail_a_policy(:meets_promotion_criteria) }
+    end
+
+    context "when the underlying setting for the upcoming change already exists in the DB (admin has modified it)" do
+      let(:promotion_status_threshold) { :beta }
+
+      before do
+        SiteSetting.enable_upload_debug_mode = false
+        DB.exec(
+          "INSERT INTO site_settings (name, value, data_type, created_at, updated_at)
+          VALUES ('enable_upload_debug_mode', 'false', 5, NOW(), NOW())",
+        )
+      end
+
+      after { DB.exec("DELETE FROM site_settings WHERE name = 'enable_upload_debug_mode'") }
+
+      it { is_expected.to fail_a_policy(:setting_not_modified) }
```

Comment:
We should wrap this example in a `context` block to say something like `when the current_change_status is not permanent`. Otherwise the example that is describing a failure is at the same level as the block describing a successful outcome.